Calculator_Project/Source.cpp

```diff
@@ -0,0 +1,251 @@
+        case '7':
+            if (equ_str_index == 0) {
+                num_str = "7";
+            }
+            else if (char_is_num(equ_str[equ_str_index - 1]) == false) {
+                num_str = "7";
+            }
+            else {
+                num_str.append("7");
+            }
+
+            if (equ_str_index == (equ_str.length() - 1)) {
+                equ_vec.push_back(num_str);
+            }
+            else if (char_is_num(equ_str[equ_str_index + 1]) == false) {
+                equ_vec.push_back(num_str);
+            }
+            break;
+        case '8':
+            if (equ_str_index == 0) {
+                num_str = "8";
+            }
+            else if (char_is_num(equ_str[equ_str_index - 1]) == false) {
+                num_str = "8";
+            }
+            else {
+                num_str.append("8");
+            }
+
+            if (equ_str_index == (equ_str.length() - 1)) {
+                equ_vec.push_back(num_str);
+            }
+            else if (char_is_num(equ_str[equ_str_index + 1]) == false) {
+                equ_vec.push_back(num_str);
+            }
+            break;
+        case '9':
+            if (equ_str_index == 0) {
+                num_str = "9";
+            }
+            else if (char_is_num(equ_str[equ_str_index - 1]) == false) {
+                num_str = "9";
+            }
+            else {
+                num_str.append("9");
+            }
+
+            if (equ_str_index == (equ_str.length() - 1)) {
+                equ_vec.push_back(num_str);
+            }
+            else if (char_is_num(equ_str[equ_str_index + 1]) == false) {
+                equ_vec.push_back(num_str);
+            }
+            break;
+        case '+':                                       //If the current character is a +,
+            equ_vec.push_back("+");                     //append it directly to the equation vector
+            break;
+        default:                                        //If the current character is anything else,
+            std::cout << "ERROR: Undefined character!"; //print error message - needs actual error handling process?
+            break;
+        }
+        equ_str_index++;        //Increment the character index tracker
+
+    }
+    vec_print<std::string>(equ_vec);        //Print equation vector for verification purposes - remove before final submission
+    std::cout << "\n\n";
+
+                                            //Start of actual calculator
+    bool found_operator = false;            //Variable to track if we completed a pass through the equation without finding an operator
+    do {
+        found_operator = false;             //Reset found operator tracker to false
+        int equ_vec_index = 0;              //Variable to track which entry in the equation we are on
+        int result;                         //Variable to store the result
+        for (std::string str : equ_vec) {   //Iterate over the entries in the equation

```

Comment:
Use `const std::string &str` here to avoid copying strings from the vector just to do a string comparison.